### PR TITLE
Update performance driver to be usable outside Pyomo

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This adds a `--project` argument tot the performance driver so that it can be directly used for projects other than Pyomo (e.g. IDAES)  Thanks to @andrewlee94 for pointing out the error.

## Changes proposed in this PR:
- Update argument processing to allow user to specift target project(s)
  - git information is collected for all projects
  - pytest is invoked "from" the first specified project (assumed to be the "main" project)
- Add some documentation

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: scripts/performance/main.py [hunk 1]
```diff
@@ -126,17 +149,18 @@ def getRunInfo(cython):
...
-    info.update(getPyomoInfo())
+    for project in options.projects:
+        info[project] = getProjectInfo(project)
     return info
 
 
-def run_tests(cython, argv):
+def run_tests(options, argv):
     gc.collect()
     gc.collect()
```
Comment (andrewlee94) [inline scripts/performance/main.py:162]: Is there a need for the duplicate `gc.collect` calls?

File: scripts/performance/main.py [hunk 2]
```diff
@@ -25,12 +26,19 @@
...
-import pyomo
-from pyomo.version import version_info as pyomo_version
 from pyomo.common.timing import TicTocTimer
 
 
 class TimingHandler(logging.Handler):
+    """Log handler to collect timing information from pyomo.common.timing
+
+    This Log Handler is attached to pyomo.common.timing and intercepts
+    the log recordsa automatically generated by Pyomo for emitting
```
Comment (blnicho) [inline scripts/performance/main.py:36]: Typo: "recordsa"

File: scripts/performance/main.py [hunk 3]
```diff
@@ -98,24 +113,32 @@ def pytest_runtest_call(self, item):
         self._timingHandler.clearTest()
 
 
-def getPyomoInfo():
+def getProjectInfo(project):
+    "Collect git and version info for a particulat project"
```
Comment (blnicho) [inline scripts/performance/main.py:117]: Typo: particular

**Review by blnicho — APPROVED** [review]
I found a couple typos but otherwise this looks fine

**Pull request merged**